src/validation_workflow/docker/validation_docker.py

```diff
@@ -0,0 +1,203 @@
+                while True:
+                    if self.check_http_request():
+                        logging.info('\n\ncluster is now ready for API test\n\n')
+                        break
+                    logging.info('sleeping 10sec')
+                    time.sleep(10)
+
+                # STEP 4 . OS, OSD API validation
+                _test_result, _counter = TestCases().test_cases()
+
+                if (_test_result):
+                    logging.info(f'All tests Pass : {_counter}')
+                    return True
+                else:
+                    raise Exception(f'Not all tests Pass : {_counter}')
+            else:
+                raise Exception('The container is fail to run. Exiting the validation.')
+        else:
+            logging.info('\n\nImage digest is Not validated. Exiting..\n\n')
+            raise Exception('Image digest does not match between the opensearchstaging at dockerHub and opensearchproject at local download. Exiting the validation.')
+
+    def check_http_request(self) -> bool:
+        try:
+            status_code, response_text = ApiTest('https://localhost:9200/').api_get()
+            if status_code == 200:
+                return True
+            else:
+                return False
+        except (requests.exceptions.ConnectionError, requests.exceptions.ConnectTimeout) as e:
+            logging.error(f'Error connecting to https://localhost:9200/: {e}')
+            return False
+
+    def cleanup(self) -> bool:
+        # clean up
+        if self.cleanup_process():
+            logging.info('cleanup is completed')
+            return True
+        else:
+            logging.info('cleanup is Not completed')
+            return False
+
+    def cleanup_process(self) -> bool:
+        # stop the containers
+        docker_command = f'{"docker-compose -f"} {self._target_yml_file} {"down"}'
+        result = subprocess.run(docker_command, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+
+        # remove docker-compose.yml from the tmp folder
+        try:
+            os.remove(self._target_yml_file)
+        except OSError as e:
+            print("Error: %s - %s." % (e.filename, e.strerror))
+
+        return('returncode=0' in (str(result)))
+
+    def is_container_daemon_running(self) -> Any:
+        check_docker_bash = "./src/validation_workflow/docker/check_docker_daemon.sh"
+        result = subprocess.call([check_docker_bash])
+        return (result)
+
+    def run_container(self, OpenSearch_image: str, OpenSearchDashboard_image: str, version: str, OpenSearchNode1_name: str, OpenSearchNode2_name: str, OpenSearchDashboard_name: str) -> Any:
+        # replace the placeholder in template docker-compose.yaml which is taken from sample in https://opensearch.org/docs/latest/opensearch/install/docker/#sample-docker-composeyml
+        yml_template = 'docker-compose.yml'
+        yml_template_location = './src/validation_workflow/docker/'
```

Comment:
Use path.join to form paths

Reply:
Thanks, @gaiksaya , got it, let us update it at next commit.